app/controllers/kudos_controller.rb

```diff
@@ -36,15 +36,18 @@ def create
     else
       respond_to do |format|
         format.html do
+          msg = "We couldn't save your kudos, sorry!"
+          commentable = @kudo.commentable
           if @kudo.dup?
-            msg = "You have already left kudos here. :)"
-          elsif @kudo.creator_of_work?
+            msg = 'You have already left kudos here. :)'
+          end
+          if @kudo.creator_of_work?
             msg = "You can't leave kudos on your own work."
-          else
-            msg = "We couldn't save your kudos, sorry!"
+          end
+          if !current_user.present? && commentable.restricted?
+            msg = "You can't leave guest kudos on a restricted work"
```

Comment:
I know it's a tiny detail, but could you put a period on these messages, so they match some of the other warnings?